### PR TITLE
Adds creamatoriums

### PR DESCRIPTION
**Pull request by QualityVan**
Like crematoriums, except they produce delicious icecream from any incinerated mobs. Adminspawn only until such time as an enterprising mapper puts them somewhere.

It turns out that all crematoriums are activated by any crematorium-activating button, incidentally.

### CODE REVIEW

**Comment by Chocobro** [comment]
<TheGhostOfWhibyl1> [tgstation] QualityVan opened pull request #28158: Adds creamatoriums (master...creamator) https://github.com/tgstation/tgstation/pull/28158
<atlanta_ned> NOPE
<atlanta_ned> CLOSE IT
<atlanta_ned> CLOSE IT
<atlanta_ned> 2LEWD

**Comment by WJohn** [comment]
you... you turn people into icecream?

for what purpose

**Comment by vcordie** [comment]
SOYLENT GREEN IS PEOPLE. PEOPLE.

**Comment by Cheridan** [comment]
this is the dumbest thing i've ever heard of

👍

**Comment by GunHog** [comment]
FUND IT

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 1]
```diff
@@ -206,8 +211,15 @@
 		if ("blue")
 			desc = "A delicious [cone_type] cone filled with blue ice cream. Made with real... blue?"
 			reagents.add_reagent("singulo", 2)
+		if ("mob")
+			desc = "A suspicious [cone_type] cone filled with bright red ice cream. That's probably not strawberry."
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm]: Add a proper ellipsis.

"That's probably not strawberry..."

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -223,6 +224,20 @@ GLOBAL_LIST_EMPTY(crematoriums)
 			update_icon()
 			playsound(src.loc, 'sound/machines/ding.ogg', 50, 1) //you horrible people
 
+/obj/structure/bodycontainer/crematorium/creamatorium
+	name = "creamatorium"
+	desc = "A human incinerator. Works well during ice cream socials."
+
+/obj/structure/bodycontainer/crematorium/creamatorium/cremate(mob/user)
+	var/list/icecreams = new()
```
Comment (AnturK) [inline code/game/objects/structures/morgue.dm]: More like icescreams.

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -223,6 +224,20 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+/obj/structure/bodycontainer/crematorium/creamatorium/cremate(mob/user)
+	var/list/icecreams = new()
+	for(var/mob/living/M in GetAllContents())
+		var/obj/item/weapon/reagent_containers/food/snacks/icecream/IC = new()
+		IC.set_cone_type("waffle")
+		IC.add_mob_flavor(M)
+		icecreams += IC
+	. = ..()
+	for(var/obj/IC in icecreams)
+		IC.loc = src
```
Comment (AnturK) [inline code/game/objects/structures/morgue.dm]: forceMove, or you could just pass the loc in creation anyway ? Don't really see the need for another loop ?

Comment (QualityVan) [inline code/game/objects/structures/morgue.dm]: After the cremator burns everything, the mobs don't exist so you can't iterate over them. Conversely, if you put the icecream in the cremator before it burns everything, it'll incinerate the icecream.

**Comment by Bawhoppen** [comment]
make the color of ice cream change based on skin color (cremate lizards for strawberry)

**Comment by Grazyn** [comment]
Can't you already make ice cream with gibs? Scoop up gibs, pour in ice cream cart?

**Comment by Bawhoppen** [comment]
Is this idea from an American Dad episode that aired recently? If not, then I have a pretty big coincidence I found.

**Pull request merged**